[Routing] Using Symfony string for routing path instead of Gedmo urlizer (#1108)
Sylius/SyliusResourceBundle: UPGRADE.md

### FROM `1.13.x` to `1.14.x`

#### Minimal Requirements

The minimum required PHP version is now 8.2.

#### Routing Path

Routing paths may have been changed depending on your configuration.

| behat/transliterator | Routing Path BC-layer | Routing paths |
|----------------------|-----------------------|---------------|
| ✅ Installed          | ✅ Enabled by default  | ✅ Unchanged   |
| ❌ Not installed      | ❌ Disabled by default | ❌ Changed     |

If you want to keep the previous paths, please follow these following steps:

> 1/ Make sure to have Behat transliterator installed.

Reviewer: ```suggestion
1. Make sure to have Behat transliterator installed.
```
Same w/ the second one 👋

```shell
composer require behat/transliterator
```

2/ Ensure routing path bc-layer is enabled

```yaml
# config/packages/
sylius_resource:
    routing_path_bc_layer: true
```

> When using the `ResourceController` for your routes and when the routing bath bc layer is enabled, the routing path

Reviewer: ```suggestion
When using the `ResourceController` for your routes and when the BC-layer is enabled, the routing path
```

contains a trailing slash by default. In 2.x, it will be removed.

> You can disable the bc-layer to remove these trailing slashes on all your routes and be prepared for 2.x area.

Reviewer: ```suggestion
You can disable the BC-layer to remove these trailing slashes on all your routes and be prepared for the 2.x release.
```

If you are using the new Routing system with `AsResource` attribute and operations, this change has no effect, there is
no trailing slash in both configuration.

```yaml
# config/packages/
sylius_resource:
    routing_path_bc_layer: false
```

| Routing system               | Routing Path BC-layer | Trailing slash |
|------------------------------|-----------------------|----------------|
| Using the ResourceController | ✅ Enabled             | ✅              |
| Using the ResourceController | ❌ Disabled            | ❌              |
| Using AsResource attribute   | ✅ Enabled             | ❌              |
| Using AsResource attribute   | ❌ Disabled            | ❌              |

The routing path uses dashes to separate words by default. Eg: `/shipping-categories`.

> On the new routing system only, you can use configure the bundle to use underscores instead. Eg: `/shipping_categories`

Reviewer: ```suggestion
On the new routing system only, you can configure using underscores instead. Eg: `/shipping_categories`
```

```yaml
# config/packages/
sylius_resource:
    path_segment_name_generator: sylius.metadata.path_segment_name_generator.underscore
```

Note that even if this is configurable, it's recommended to use dashes to separate the words for SEO.

*Source:*
* https://developers.google.com/search/docs/crawling-indexing/url-structure

## UPGRADE FOR `1.13.x`

### FROM `1.12.x` to `1.13.x`

> The `Sylius\Resource\Exception\VariantWithNoOptionsValuesException` and

Author: I may have applied a reformat code...
